src/rpc/rpc.cc

```diff
@@ -69,10 +69,26 @@ RPCStatus SendRPCMessage(const RPCMessage& msg, const int32_t target_id) {
 }
 
 RPCStatus RecvRPCMessage(RPCMessage* msg, int32_t timeout) {
-  // ignore timeout now
-  CHECK_EQ(timeout, 0) << "rpc cannot support timeout now.";
-  RPCContext::getInstance()->receiver->Recv(msg);
-  return kRPCSuccess;
+  static constexpr int32_t default_timeout = 5 * 1000;  // milliseconds
+  RPCStatus status;
+  const int32_t real_timeout = timeout == 0 ? default_timeout : timeout;
+  do {
+    status = RPCContext::getInstance()->receiver->Recv(msg, real_timeout);
+    if (status == kRPCTimeOut) {
+      static const std::string log_str = [real_timeout, timeout]() {
```

Comment:
The code is strange. Why defining a lambda?

Reply:
in order to run once only.